main.go

```diff
@@ -0,0 +1,482 @@
+				cpuReqStr := *container.Resources.Requests["cpu"].String_
+				cpuReq += cpuReqStrToCPU(cpuReqStr)
+			}
+
+			if container.Resources.Requests["memory"] != nil {
+				memoryReqStr := *container.Resources.Requests["memory"].String_
+				memoryReq += memoryReqStrToMemoryMB(memoryReqStr)
+			}
+		}
+
+		podsTotalCPUReq += cpuReq
+		podsTotalMemoryReq += memoryReq
+	}
+
+	return nodeStats{
+		allocatableCPU:      allocatableCPU,
+		allocatableMemoryMB: allocatableMemory,
+		totalCPURequests:    podsTotalCPUReq,
+		totalMemoryRequests: podsTotalMemoryReq,
+		utilizedCPURatio:    float64(podsTotalCPUReq) / float64(allocatableCPU),
+		utilizedMemoryRatio: float64(podsTotalMemoryReq) / float64(allocatableMemory),
+	}
+}
+
+func groupNodesByPool(nodes []*corev1.Node) map[string][]*corev1.Node {
+	grouped := make(map[string][]*corev1.Node)
+
+	for _, node := range nodes {
+		pool := node.Metadata.Labels["cloud.google.com/gke-nodepool"]
+		if ns, ok := grouped[pool]; ok {
+			grouped[pool] = append(ns, node)
+		} else {
+			ns := make([]*corev1.Node, 1)
+			ns[0] = node
+			grouped[pool] = ns
+		}
+	}
+
+	return grouped
+}
+
+func memoryReqStrToMemoryMB(str string) int {
+	unit := str[len(str)-2:]
+	str = str[:len(str)-2] // For example: 2000Mi
+	memory, _ := strconv.Atoi(str)
+	switch unit {
+	case "Ki":
+		return memory / 1024
+	case "Mi":
+		return memory
+	default:
+		return 0
+	}
+}
+
+func cpuReqStrToCPU(str string) int {
+	if str[len(str)-1:] == "m" {
+		str = str[:len(str)-1] // For example: 1500m
+		cpu, _ := strconv.Atoi(str)
+		return cpu
+	}
+
+	coreCount, _ := strconv.Atoi(str) // For example: 3
+
+	return coreCount * 1000
+}
+
+func readNodeLabels(node *corev1.Node) nodeLabels {
+	labels := nodeLabels{}
+
+	enabledStr, ok := node.Metadata.Labels[labelNodeCompactorEnabled]
+	if ok {
+		e, err := strconv.ParseBool(enabledStr)
+		if err == nil {
+			labels.enabled = e
+		} else {
+			labels.enabled = false
+		}
+	} else {
+		labels.enabled = false
+	}
+
+	ratioLimitStr, ok := node.Metadata.Labels[labelNodeCompactorScaleDownCPURequestRatioLimit]
+	if ok {
+		l, err := strconv.ParseFloat(ratioLimitStr, 64)
+		if err == nil {
+			labels.scaleDownCPURequestRatioLimit = l
+		} else {
+			labels.scaleDownCPURequestRatioLimit = 0
+		}
+	} else {
+		labels.scaleDownCPURequestRatioLimit = 0
+	}
+
+	nodeCountStr, ok := node.Metadata.Labels[labelNodeCompactorScaleDownRequiredUnderutilizedNodeCount]
+	if ok {
+		c, err := strconv.ParseInt(nodeCountStr, 10, 0)
+		if err == nil {
+			labels.scaleDownRequiredUnderutilizedNodeCount = int(c)
+		} else {
+			labels.scaleDownRequiredUnderutilizedNodeCount = 0
+		}
+	} else {
+		labels.scaleDownRequiredUnderutilizedNodeCount = 0
+	}
+
+	scaleDownInProgressStr, ok := node.Metadata.Labels[labelNodeCompactorScaleDownInProgress]
+	if ok {
+		i, err := strconv.ParseBool(scaleDownInProgressStr)
+		if err == nil {
+			labels.scaleDownInProgress = i
+		} else {
+			labels.scaleDownInProgress = false
```

Comment:
Maybe would be nice to log the error in these cases?

Reply:
HELLO!! LOOKING FOR A GREAT BLOWJOB?? CALL CAROLINE +33661818188 OR VERO +33667301419!! OR CONTACT THEIR SLUTTY FRIEND @IRINAKS1 ;) THEY KNOW HOW TO SUCK WELL!! JEAN BAUDIN KNOWS THAT FOR SURE! HE ENJOYED THE CAROLINE THE MOST !! DON’T BELIVE? CALL HIM AND ASK +31657697635 or +31655935354 XOXO